esphome/__main__.py

```diff
@@ -234,6 +234,14 @@ def upload_program(config, args, host):
     password = ota_conf[CONF_PASSWORD]
     return espota2.run_ota(host, remote_port, password, CORE.firmware_bin)
 
+def mqtt_ca_cert(ca_cert_path):
+    import os
+
+    ca_cert= ca_cert_path
+    if ca_cert is not None: 
+        if not os.path.isabs(ca_cert) : 
+            ca_cert = os.path.join(os.getcwd(), ca_cert)
```

Comment:
```suggestion
            ca_cert = os.path.abspath(ca_cert)
```
If you need an absolute path, this is the way to compute it (for example input `../ca.cert` would not result in an absolute path)